### ISSUE TITLE
Boundingbox on Link API

### ISSUE THREAD

**Issue by jfelixre** [gazebosim/gz-sim#2353]
There is an alternative of Gazebo-classic's Link API Boundingbox function for ignition?

API reference page still shows a "TODO"

thanks!

**Comment by azeey** [comment]
We already have the ability to get the bounding box, but it's through a component. Documenting this would be a good step as well.

**Comment by gabrielfpacheco** [comment]
> We already have the ability to get the bounding box, but it's through a component. Documenting this would be a good step as well.

@azeey, isn't that only available to get the model bounding box, since the Physics plugin set it [here](https://github.com/gazebosim/gz-sim/blob/gz-sim9/src/systems/physics/Physics.cc#L2887-L2892)?

I couldn’t find any source that actually sets a `gz::components::AxisAligned` for a link entity. Therefore, as I had to get the bounding box of the link for a system plugin I developed, I created #2778 and https://github.com/gazebosim/sdformat/pull/1540 to provide such an API.

Please let me know if I'm missing something, I hope this could be helpful to others.

**Comment by iche033** [comment]
I think the alternative approach would be to update the the physics system populate the bounding box component for links as well? I see the API available in gz-physics, [GetLinkBoundingBox](https://github.com/gazebosim/gz-physics/blob/gz-physics8/include/gz/physics/GetBoundingBox.hh#L40)

**Comment by gabrielfpacheco** [comment]
> I think the alternative approach would be to update the the physics system populate the bounding box component for links as well

Yes, I think this would probably be more straightforward to implement. If I understood it correctly, this feature should be available for all currently supported physics engines, right?

I initially thought not to update the physics plugin because I did not want to add extra load to a critical plugin if this was not an actual requirement. However, I believe we could the same as for the model and only update the link entities containing the corresponding components.

Maybe I could update https://github.com/gazebosim/gz-sim/pull/2787 to add this feature to the physics system and update the link API so that it only computes from the SDF if an `AxisAlignedBox` component does not exist for the link entity. Or I could completely remove the calculation via SDF if you think it is more adequate.

Please, let me know what would be the best path forward.

**Comment by iche033** [comment]
I think getting the bounding box info from the physics system is preferred because it has the information already, and there is no need for a custom mesh bounding box calculator. In the rare case the the users does not load a physics system in Gazebo (or a particular physics engine does not support the bounding box feature), we could then fall back to using the SDF API.

I think the first step is to implement a [similar logic for populating a model AxisAlignedBox](https://github.com/gazebosim/gz-sim/blob/7eef2303c6096bb8b547439239e81d5c8571234e/src/systems/physics/Physics.cc#L2855-L2895) for links in the physics system. The expected usage here is that plugin writers would need to manually create the `AxisAlignedBox` component for a Link, the physics system then populates it.

We can then update the Link class to add the `AxisAlignedBox` API like you did in https://github.com/gazebosim/gz-sim/pull/2787, except we would be the one creating the `AxisAlignedBox` component for the link in that class. I think this could be done through an `EnableBoundingBoxChecks` call (similar to `EnableVelocityChecks`). 

If you're ok with this approach, could you do the changes in the physics system in a separate PR? Mainly just to make the review process easier. Thanks for the contributions so far!

**Comment by gabrielfpacheco** [comment]
> If you're ok with this approach, could you do the changes in the physics system in a separate PR? Mainly just to make the review process easier

Yeah, I like the suggested approach. As soon as I have some free cycles, I will make the changes to the open PR and open a new one for the physics.

> Thanks for the contributions so far!

My pleasure :)
